src/decoder/stream.rs

```diff
@@ -68,9 +68,21 @@ pub enum Decoded {
     ImageEnd,
 }
 
+/// Any kind of error during PNG decoding.
+///
+/// This enumeration provides a very rough analysis on the origin of the failure. That is, each
+/// variant corresponds to one kind of actor causing the error. It should be understood as a direct
```

Comment:
Should this be "It *shouldn't* be ..." ?

Reply:
That is correct.